Sources/SmithyXML/Reader/Reader.swift

```diff
@@ -5,18 +5,22 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+import protocol SmithyReadWrite.SmithyReader
+import enum SmithyReadWrite.Document
 import typealias SmithyReadWrite.ReadingClosure
 import struct Foundation.Date
 import struct Foundation.Data
 import enum SmithyTimestamps.TimestampFormat
 import struct SmithyTimestamps.TimestampFormatter
```

Comment:
Made to conform to `SmithyReader`.  Removed several functions that are now provided by protocol extensions.

Also made the reader smarter about whether it corresponds to content.